### ISSUE TITLE
InvalidLinkBearTest: timeout tests are invalid

### ISSUE THREAD

**Issue by jayvdb** [coala/coala-bears#1240]
`test_variable_timeouts` uses setting `timeout`, which is not valid as the setting in the bear is called `network_timeout`.

Also it isnt clear that these tests actually do anything especially useful, particularly not anything to do with timeouts as the timeout value is not being used.

**Comment by jayvdb** [comment]
`requests_mock` does support timeouts being set, but doesnt attempt to do a delay, but that would at least confirm the value in the Setting is making it through to the `requests` call.

getsentry also doesnt support it: https://github.com/getsentry/responses/pull/115

https://httpbin.org/delay/3 should work; and can probably be done without network using `pytest-httpbin`.

**Comment by sils** [comment]
it uses [network_timeout](https://github.com/coala/coala-bears/pull/877/commits/f72c246f9967d498660668512cd8f99eae7589fe#diff-c4521389f8a26d35c040551df14b3557R257) and it asserts that timeout is called through the mock.

**Comment by jayvdb** [comment]
Ah I see.  The use of `return_value=response(status_code=200)` is confusing, as that definitely isnt testing a timeout, and a inner func `response` isnt needed as the return_value is constant for all mocked call.

But I see now the tests are just verifying the setting is passed through, and are doing that ok, so I've updated the title.

We need mocked or httpbin tests for what happens when there is a timeout.

**Comment by Mixih** [comment]
So a proper timeout test? The risk of that is that it will hold up the test until the link times out, unless we the httpbin with timeout.

**Comment by meetmangukiya** [comment]
Yeah, the timeout is passed correctly, checked via `pdb` and also the tests verify it, I think we can rely on `requests` to take care of the rest, being a popular library :/

**Comment by Makman2** [comment]
imo current tests do suffice :+1:

**Comment by jack17529** [comment]
@jayvdb is the issue over ?
